### PR TITLE
fix: move image decoding logic from bubble item to notify entity

### PR DESCRIPTION
**Pull request by xujin177**
The image decoding functions (copyLineRGB32, copyLineARGB32, decodeImageFromDBusArgument, decodeImageToBase64,
imagePathOfNotification) and the appIcon resolution logic have been moved from BubbleItem to NotifyEntity. This refactoring centralizes image processing in the entity layer, reducing code duplication and enabling reuse of the resolved icon logic in the NotificationManager. The new appIconResolved() method replaces the local logic in BubbleItem::appIcon(), and the manager now pre-sets the resolved icon on the entity before further processing.

Log: Refactored notification image handling by moving image decoding from bubble UI to notification entity

Influence:
1. Verify that notification icons from image-data/icon_data hints are correctly displayed in bubble popups
2. Test notification icons from appIcon fallback when no hints are provided
3. Verify that the icon resolution in NotificationManager does not affect existing notification behavior
4. Test various image formats (RGB32, ARGB32) and edge cases (invalid data, incomplete images)
5. Verify that base64 data URIs from hints are preserved and displayed correctly
6. Test with applications that send desktop-entry hints and verify icon lookup

fix: 将图片解码逻辑从气泡项迁移到通知实体

将图片解码函数（copyLineRGB32、copyLineARGB32、
decodeImageFromDBusArgument、decodeImageToBase64、
imagePathOfNotification）以及应用图标解析逻辑从 BubbleItem 移动
到 NotifyEntity。此重构将图片处理集中在实体层，减少代码重复，并使
NotificationManager 能够重用已解析的图标逻辑。新增的 appIconResolved() 方法替代了 BubbleItem::appIcon() 中的本地逻辑，管理器现在在进一步处理前
预先设置已解析的图标到实体。

Log: 重构通知图片处理，将图片解码从气泡 UI 移至通知实体

Influence:
1. 验证来自 image-data/icon_data 提示的通知图标在气泡弹窗中正确显示
2. 测试未提供提示时使用 appIcon 回退的通知图标
3. 验证 NotificationManager 中的图标解析不影响现有通知行为
4. 测试各种图片格式（RGB32、ARGB32）及边界情况（无效数据、不完整图片）
5. 验证来自提示的 base64 数据 URI 被正确保留和显示
6. 测试发送 desktop-entry 提示的应用，验证图标查找功能

PMS: BUG-370701

### CODE REVIEW

File: panels/notification/bubble/bubbleitem.cpp [hunk 1]
```diff
@@ -195,16 +63,7 @@ QString BubbleItem::appName() const
...
-    const QString iconFromHints = imagePathOfNotification(m_entity.hints(), m_entity.appIcon(), m_entity.appName());
-    if (!iconFromHints.isEmpty())
-        return iconFromHints;
-
-    if (!m_entity.appIcon().isEmpty()) {
-        return m_entity.appIcon();
-    }
-
-    return {};
+    return m_entity.appIconResolved();
```
Comment (18202781743) [inline panels/notification/bubble/bubbleitem.cpp:66]: 既然在[notificationmanager.cpp](https://github.com/linuxdeepin/dde-shell/pull/1671/changes#diff-5505adbaf5b86cdc1c7894e91be8d607cffc368b65d157dec84d666372b6b833) 里设置了appIcon，这里为什么还要请求一次呀，另外appIconResolved这个可以做成静态的函数，

Comment (xujin177) [reply panels/notification/bubble/bubbleitem.cpp:66]: 已根据评论做相应修改~

File: panels/notification/server/notificationmanager.cpp [hunk 2]
```diff
@@ -259,6 +259,11 @@ uint NotificationManager::Notify(const QString &appName, uint replacesId, const
     entity.setProcessedType(NotifyEntity::None);
     entity.setReplacesId(replacesId);
 
+    const auto resolvedIcon = NotifyEntity::appIconResolved(entity.hints(), entity.appIcon());
```
Comment (18202781743) [inline panels/notification/server/notificationmanager.cpp]: 这个要不不在这里保存吧，它会修改entity里实际的icon，这个会存储在数据库里，这样的话，原来的icon丢失了，另外处理的icon可能又重复保存了，可能hints里有一份，这个icon里也有一份，

**Review by 18202781743 — DISMISSED** [review]

**Review by 18202781743 — APPROVED** [review]
